### PR TITLE
增加GPT修复功能

### PR DESCRIPTION
**Pull request by CHNWangXi**
在EMMC界面为维修人员增加了GPT修复的功能，用于为新EMMC快捷的添加GPT分区，同时兼容32/64G

### CODE REVIEW

**Comment by CTCaer** [comment]
Thanks for the PR but it won't be accepted.

The lesser stuff:
- No non-english descriptions and comments. Translators exist for that reason.
- You are doing weird things. Heap and memcpy are a thing. Also you are using wrong names and wrong styling.
- You are using the GPT and table as binaries where you should just properly define it. I already provide structs for that.

The important stuff (reason for closing it):
What you PRed basically already exists, both on PC and on switch payloads.
And these do it correctly (you missed the backup GPT, without it the backup partition table, which you call gptSuffix is useless).
You also assume size for the else case which can be catastrophic.

If you still consider to PR, a more useful feature would be this:
Automatic check of existing table and repair.
If missing offer a dynamic init of it.
In case it's missing or all are good, offer this:
Init as HOS or L4T or Android only.
Init as combo of the above.
If GPT is fine and there's empty space (user restored a backup to a modded bigger eMMC), offer to add L4T or Android or both.
If OK offer a repartition of GPT USER partition, while informing of formatting and addition of the aforementioned partitions.
Basically a copy paste of partition manager, edited for eMMC.

Another thing would be to have a proper dynamic repair/init stuff.
For example user partitioned it in Linux but GPT is make HOS throw purple screen because empty partitions.
Offer repair of that, aka fix entries num and header crc and done.

Anyway, as it is now, it's an improper implementation of already existing trivial stuff.
